src/ServiceControl/Recoverability/Retrying/Infrastructure/ReturnToSender.cs

```diff
@@ -2,22 +2,29 @@ namespace ServiceControl.Recoverability
 {
     using System.Collections.Generic;
     using System.IO;
+    using System.Linq;
+    using System.Text;
     using System.Threading.Tasks;
+    using CompositeViews.Messages;
+    using MessageFailures;
     using NServiceBus.Logging;
     using NServiceBus.Routing;
     using NServiceBus.Transport;
     using Operations.BodyStorage;
+    using Raven.Client;
+    using Raven.Json.Linq;
 
     class ReturnToSender
     {
-        public ReturnToSender(IBodyStorage bodyStorage)
+        public ReturnToSender(IBodyStorage bodyStorage, IDocumentStore documentStore)
```

Comment:
Would it make sense to hide some of the implementation details behind `IBodyStorage`? Or to create a new abstration that encompasses both of these approaches. There's a lot of body retrieval code in the return to sender now.

Reply:
I'm happy to address this once we have some good data that shows this approach is worth doing.

Reply:
we also already have similar yet different logic inside the `GetBodyByIdApi` controller

Reply:
Mike for now I refactored it into two methods. I'm wondering how much more it should be abstracted since the GetBodyByIdApi uses a different mechanism to fetch the body. I could try to see if we can use the same ID based querying with transformer there as a quick spike

Reply:
The API needs to query by by the message id of the last attempt for some reasons and uses 

```
               var message = await session.Query<MessagesViewIndex.SortAndFilterOptions, MessagesViewIndex>()
                    .Statistics(out var stats)
                    .TransformWith<MessagesBodyTransformer, MessagesBodyTransformer.Result>()
                    .FirstOrDefaultAsync(f => f.MessageId == messageId)
                    .ConfigureAwait(false);
```

I don't think we can consolidate that